### PR TITLE
Add abstracts and inline TOCs, and document Cypher9 coverage

### PR DESCRIPTION
**Pull request by jjaderberg**

### CODE REVIEW

File: documentation/asciidoc/cypher-cypher9-features.adoc [hunk 1]
```diff
@@ -1,15 +1,229 @@
...
+| Boolean       | {yes}
+| Date          | {yes}
+| DateTime      | {no}
+| Duration      | {yes}
+| Float         | {yes}
+| Integer       | {yes}
+| List          | {yes}
+| LocalDateTime | {yes}
+| LocalTime     | {no}
+| Map           | {no}
```
Comment (s1ck) [inline documentation/asciidoc/cypher-cypher9-features.adoc]: we support maps, see `CTMap`

File: documentation/asciidoc/cypher-cypher9-features.adoc [hunk 2]
```diff
@@ -1,15 +1,229 @@
...
+|===
+| name                   | status  | name | status
+| CASE                   | {yes}
+| all                    | {no}
+| allShortestPaths       | {no}
+| any                    | {no}
+| extract                | {never}
+| filter                 | {no}
+| list comprehension     | {no}
+| map projections        | {no}
```
Comment (s1ck) [inline documentation/asciidoc/cypher-cypher9-features.adoc]: What is an example for map projections? Something like `RETURN { foo : a.bar }`? If yes, then it's a `yes`.

File: documentation/asciidoc/cypher-cypher9-features.adoc [hunk 3]
```diff
@@ -1,15 +1,229 @@
...
+| AND                   | {yes}
+| CONTAINS              | {yes}
+| DISTINCT              | {yes}
+| ENDS WITH             | {yes}
+| IS NOT NULL           | {yes}
+| IS NULL               | {yes}
+| NOT                   | {yes}
+| OR                    | {yes}
+| STARTS WITH           | {yes}
+| XOR                   | {no}
```
Comment (s1ck) [inline documentation/asciidoc/cypher-cypher9-features.adoc]: we don't? woah, I love that op

File: documentation/asciidoc/index.adoc [hunk 4]
```diff
@@ -3,10 +3,20 @@
 
 [abstract]
 --
-Placeholder abstract.
+This part provides comprehensive documentation for Cypher for Apache Spark.
```
Comment (s1ck) [inline documentation/asciidoc/index.adoc]: Is there a reason for not using `CAPS` instead?

File: documentation/asciidoc/cypher-cypher9-features.adoc (hunk 2 above)
Comment (jjaderberg) [reply documentation/asciidoc/cypher-cypher9-features.adoc]:
```
MATCH (actor:Person { name: 'Charlie Sheen' })-[:ACTED_IN]->(movie:Movie)
RETURN actor { .name, .realName, movies: collect(movie { .title, .year })}
```
https://neo4j.com/docs/cypher-manual/current/syntax/maps/#cypher-map-projection

Comment (s1ck) [reply documentation/asciidoc/cypher-cypher9-features.adoc]: haha, a `DesugaredMapProjection` ... never saw that before but it looks neat

File: documentation/asciidoc/backend-sql-graphddl.adoc [hunk 5]
```diff
@@ -177,9 +173,9 @@ CREATE ELEMENT TYPE MyType (
 )
 ----
 
-Possible data types reflect a subset of the Cypher type system. 
+Possible data types reflect a subset of the Cypher type system.
 Currently supported types are:
-`STRING`, `INTEGER`, `FLOAT` and `BOOLEAN` as well as their nullable counterparts, i.e. `STRING?`, `INTEGER?`, `FLOAT?` and `BOOLEAN?`.
+`STRING`, `INTEGER`, `FLOAT` and `BOOLEAN` as well as their nullable counterparts, i.e., `STRING?`, `INTEGER?`, `FLOAT?` and `BOOLEAN?`.
```
Comment (s1ck) [inline documentation/asciidoc/backend-sql-graphddl.adoc]:
```suggestion
`STRING`, `INTEGER`, `FLOAT`, `BOOLEAN` and `DATE` as well as their nullable counterparts, i.e., `STRING?`, `INTEGER?`, `FLOAT?`, `BOOLEAN?` and `DATE?`.
```

File: documentation/asciidoc/backend-sql.adoc [hunk 6]
```diff
@@ -42,13 +42,26 @@ The above Graph DDL requires two tables (with columns): `view_KNOWS (start_id, e
...
 
-* The referenced tables are required to contain all the columns referenced in the Graph DDL, but may also contain additional columns; these will not be selected.
-* By default, there is a case-insensitive mapping between the label properties to table columns, such that a table `foo(BAR1 varchar, BAR2 bigint)` will map into a label `(Foo {bar1: STRING, BAR2: INTEGER})`.
-  The case in the graph will follow the case specified in the graph schema, as Cypher is case sensitive when it comes to property keys.
+The referenced tables are required to contain all the columns referenced in the Graph DDL.
+The tables may contain additional columns not referenced in the Graph DDL.
+Such additional columns will not be selected.
+
+Mappings are not case sensitive.
+A table `person(NAME varchar, AGE bigint)` can be mapped into a label `(Person {name: STRING, age: INTEGER})`.
```
Comment (s1ck) [inline documentation/asciidoc/backend-sql.adoc]:
```suggestion
A table `person(NAME varchar, AGE bigint, HOMETOWN varchar)` can be mapped into a label `(Person {name: STRING, age: INTEGER, homeTown: String})`.
```

File: documentation/asciidoc/backend-sql.adoc [hunk 7]
```diff
@@ -42,13 +42,26 @@ The above Graph DDL requires two tables (with columns): `view_KNOWS (start_id, e
...
+The resulting graph will follow the case specified in the schema.
+Cypher is case sensitive.
+To query a graph created with this mapping, the label and property names used in the query must match the case in the graph:
 
+[source, cypher]
+----
+MATCH (p:Person)
+WHERE p.name = 'Tobias'
```
Comment (s1ck) [inline documentation/asciidoc/backend-sql.adoc]:
```suggestion
WHERE p.name = 'Tobias' AND p.homeTown = 'Malmö'
```

**Pull request merged**